### PR TITLE
Bolt Action | FE | Refactor Bolt Action components

### PR DESCRIPTION
**Pull request by danielamorse**
## Jira

- http://vjira2:8080/browse/BDS-1869
- http://vjira2:8080/browse/BDS-799

## Summary

Refactor Bolt Action components to better leverage what's available in the base class and clean up some component code.

## Details

- Move `bolt-button-reset()` mixin out of the Button component into tools where it can be used by other components like Link and Trigger.
- In `BoltAction`, remove unused `html` import and unnecessary `this.useShadow` assignment, check for `super.rendered` before calling it.
- Set `delegateFocus` to `true` by default in BoltAction and remove `delegateFocus` from components.
- Reference props directly on `BoltAction` class rather than duplicating them.
- Simplify `innerSlots` code in Link, Chip, and Button components. We want to move away from the switch statement pattern for slots which is less readable than simply returning the template in one chunk.
- Rework Button slotted content so that empty spans with `is-empty` class are not output.

## How to test

- Review the code changes.
- Review the Link, Button, Chip, and Trigger components in Chrome, IE, and Edge for regressions.

### CODE REVIEW

**Comment by danielamorse** [comment]
> This has introduced double focus outlines for buttons, we need to remove the browser default outline.

Good catch @mikemai2awesome. Now fixed. Ready for re-review.

**Comment by sghoweri** [comment]
@danielamorse could you take a peek at the failing Travis tests with this when you have a sec?

**Comment by danielamorse** [comment]
> @danielamorse could you take a peek at the failing Travis tests with this when you have a sec?

Fixed 👍

File: packages/components/bolt-chip/src/chip.js [hunk 1]
```diff
@@ -50,39 +49,19 @@ class BoltChip extends BoltAction {
...
-
-    const innerSlots = [
-      slotMarkup('before'),
-      slotMarkup('default'),
-      slotMarkup('after'),
-    ];
+    // 1. Remove line breaks before and after lit-html template tags, causes unwanted space inside and around inline links
+    // 2. Zero Width No-break Space (&#xfeff;) is needed to make the last word always stick with the icon, so the icon will never become an orphan.
+    // prettier-ignore
+    const innerSlots = html`${
```
Comment (sghoweri) [inline packages/components/bolt-chip/src/chip.js]: +1 — I really like this new pattern. Much cleaner + much simpler to wrap your head around

File: packages/components/bolt-button/src/button.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ class BoltButton extends BoltAction {
   static is = 'bolt-button';
 
   static props = {
```
Comment (sghoweri) [inline packages/components/bolt-button/src/button.js]: lol — I ended up doing pretty much this exact same thing on #1526 😝

File: packages/core/elements/bolt-action/index.js [hunk 3]
```diff
@@ -21,8 +21,8 @@ class BoltAction extends withLitHtml() {
   // https://github.com/WebReflection/document-register-element#upgrading-the-constructor-context
   constructor(self) {
     self = super(self);
-    self.useShadow = hasNativeShadowDomSupport;
     self.rootElementTags = [];
+    self.delegateFocus = true;
```
Comment (sghoweri) [inline packages/core/elements/bolt-action/index.js]: I could use your eyes on #1526 (either before or after this PR gets merged in) to make sure these specific updates to Bolt Action carry over

File: packages/core/styles/02-tools/tools-button-reset/_tools-button-reset.scss [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+/// @param {string} $value
+/// @example scss - bolt-button-reset mixin
+/// .element {
+///    @include bolt-button-reset;
+/// }
+@mixin bolt-button-reset() {
+  @include bolt-padding(0);
+  @include bolt-font-family(body);
+
+  appearance: none; // override default [type=submit] button styles coming from Normalize
```
Comment (sghoweri) [inline packages/core/styles/02-tools/tools-button-reset/_tools-button-reset.scss]: +1

File: packages/core/elements/bolt-action/index.js (hunk 3 above)
Comment (danielamorse) [reply packages/core/elements/bolt-action/index.js]: Will do!

**Comment by sghoweri** [comment]
PR was released with v2.11.0

**Pull request merged**